File: bdd/app/catalog.rb
class Catalog
  def initialize
    @products = []
  end

  def add(name, price)
    @products << Product.new(name, price)
  end

  def find(name)
    @products.find{ |p| p.name == name}
  end
end

class Product
  attr_reader :name, :price
  def initialize(name, price)
    @name = name
    @price = price
  end
end

File: behave/app/catalog.py
class Catalog(object):
  def __init__(self):
    self.products = []

  def add(self, name, price):
    self.products.append(Product(name, price))

  def find(self, name):
    for product in self.products:
      if product.name == name:
        return product
    return Product("unknown product", 0)

class Product(object):
  def __init__(self, name, price):
    self.name = name
    self.price = price